src/pay/batch_payer.py

```diff
@@ -292,13 +347,26 @@ def attempt_single_batch(self, payment_records, op_counter, dry_run=None):
             storage = self.storage_limit
             pymnt_amnt = payment_item.amount  # expects in micro tezos
 
+            # TRD extension for non scriptless contract accounts
+            gas_limit, tx_fee = self.gas_limit, self.default_fee
+            if payment_item.paymentaddress.startswith('KT'):
+                simulation_results = self.simulate_single_operation(payment_item, pymnt_amnt, branch, chain_id)
+                gas_limit, tx_fee, storage = simulation_results
+                burn_fee = COST_PER_BYTE * storage
+                total_fee = tx_fee + burn_fee
+                # Bound the total (baker and burn) fee by the reward amount in case of KT1 accounts
+                if total_fee > pymnt_amnt:
```

Comment:
where is the connection to the limits set in the fee.ini?

Reply:
The initial fee value is taken from fee.ini. The extra fee needed for gas + extra burn fee needed for storage are simulated and adjusted according to the simulation output. The values for tz1 accounts are taken directly from fee.ini.

Reply:
the boundary if(fee > reward) is not sufficient. we need maximum limits services are willing to pay. Idea: create the fee.ini at first startup. Add the name of the current protocol. check it at startup and update with new fees if protocol changes. Store the simulated values for a tz account and add max_fees to the ini as a fail criteria. The Kt accounts with internal calls can be used to trick services into paying large amounts of fees.